src/order/bounds.lean

```diff
@@ -868,36 +871,32 @@ lemma map_bdd_below : (lower_bounds s ∩ t).nonempty → bdd_below (f '' s) :=
 λ ⟨C, hs, ht⟩, ⟨f C, Hf.mem_lower_bounds_image Hst hs ht⟩
 
 /-- A monotone map sends a least element of a set to a least element of its image. -/
-lemma map_is_least (Ha : is_least s a) : is_least (f '' s) (f a) :=
-⟨mem_image_of_mem _ Ha.1, Hf.mem_lower_bounds_image Hst Ha.2 (Hst Ha.1)⟩
+lemma map_is_least (Ha : is_least t a) : is_least (f '' t) (f a) :=
+⟨mem_image_of_mem _ Ha.1, Hf.mem_lower_bounds_image_self Ha.2 Ha.1⟩
 
 /-- A monotone map sends a greatest element of a set to a greatest element of its image. -/
-lemma map_is_greatest (Ha : is_greatest s a) : is_greatest (f '' s) (f a) :=
-⟨mem_image_of_mem _ Ha.1, Hf.mem_upper_bounds_image Hst Ha.2 (Hst Ha.1)⟩
-
-lemma is_lub_image_le (Ha : is_lub s a) (Hat : a ∈ t) {b : β} (Hb : is_lub (f '' s) b) :
-  b ≤ f a :=
-Hb.2 (Hf.mem_upper_bounds_image Hst Ha.1 Hat)
-
-lemma le_is_glb_image (Ha : is_glb s a) (Hat : a ∈ t) {b : β} (Hb : is_glb (f '' s) b) :
-  f a ≤ b :=
-Hb.2 (Hf.mem_lower_bounds_image Hst Ha.1 Hat)
+lemma map_is_greatest (Ha : is_greatest t a) : is_greatest (f '' t) (f a) :=
+⟨mem_image_of_mem _ Ha.1, Hf.mem_upper_bounds_image_self Ha.2 Ha.1⟩
 
 end monotone_on
 
 namespace antitone_on
 
 variables [preorder α] [preorder β] {f : α → β} {s t : set α}
   (Hf : antitone_on f t) {a : α} (Hst : s ⊆ t)
-include Hf Hst
+include Hf
 
-lemma mem_upper_bounds_image (Has : a ∈ lower_bounds s) (Hat : a ∈ t) :
-  f a ∈ upper_bounds (f '' s) :=
-Hf.dual_right.mem_lower_bounds_image Hst Has Hat
+lemma mem_upper_bounds_image (Has : a ∈ lower_bounds s) : a ∈ t → f a ∈ upper_bounds (f '' s) :=
+Hf.dual_right.mem_lower_bounds_image Hst Has
 
-lemma mem_lower_bounds_image (Has : a ∈ upper_bounds s) (Hat : a ∈ t) :
-  f a ∈ lower_bounds (f '' s) :=
-Hf.dual_right.mem_upper_bounds_image Hst Has Hat
+lemma mem_upper_bounds_image_self : a ∈ lower_bounds t → a ∈ t → f a ∈ upper_bounds (f '' t) :=
+Hf.dual_right.mem_lower_bounds_image_self
```

Comment:
I prefer the old statements with named arguments instead of implications.

Reply:
This was done as a basic golfing measure. I can't see much benefit in having the named arguments, but I'll revert if you want.

Reply:
I like it with the implications, because it ensures that the argument order is the same, which is a desirable property of the API.

Reply:
Yeah, on further thought, I agree with Yaël that this. Making sure the API between similar statements match is a desirable thing that's easy to overlook.

Reply:
Named arguments are nice because then the argument names found by metaprogram aren't junk



> I like it with the implications, because it ensures that the argument order is the same, which is a desirable property of the API.

You get that with
```
lemma mem_upper_bounds_image_self (hl: a ∈ lower_bounds t) (ha : a ∈ t) : f a ∈ upper_bounds (f '' t) :=
Hf.dual_right.mem_lower_bounds_image_self hl ha
```
though too

Reply:
Except that it doesn't ensure it, because
```
lemma mem_upper_bounds_image_self (ha : a ∈ t) (hl: a ∈ lower_bounds t) : f a ∈ upper_bounds (f '' t) :=
Hf.dual_right.mem_lower_bounds_image_self hl ha
```
is valid as well.

Reply:
I guess 
```lean
lemma mem_upper_bounds_image_self : ∀ (hl: a ∈ lower_bounds t) (ha : a ∈ t), f a ∈ upper_bounds (f '' t) :=
Hf.dual_right.mem_lower_bounds_image_self
```
addresses my concerns, although I'll admit its seemingly not idiomatic

Reply:
The [style guide](https://leanprover-community.github.io/contribute/style.html#hypotheses-left-of-colon) does state that the preferred style is that hypotheses are to the left of the colon (and I also like that better).

But I don't care about it too much, and the rest of the PR looks good.

bors merge